common/flatpak-dir.c

```diff
@@ -6990,6 +7009,12 @@ flatpak_dir_deploy (FlatpakDir          *self,
   if (application_runtime)
     g_variant_builder_add (&metadata_builder, "{s@v}", "runtime",
                            g_variant_new_variant (g_variant_new_string (application_runtime)));
+  if (old_deploy_data)
+    {
+      g_autofree const char **previous_ids = flatpak_deploy_data_get_previous_ids (old_deploy_data, NULL);
+      g_variant_builder_add (&metadata_builder, "{s@v}", "previous-ids",
```

Comment:
We should not add the previous-ids field to the new deploy file if it is not present in the old one, so this should be conditional.